.gitignore

```diff
@@ -2,3 +2,4 @@ cve-scan*.json
 sbom-spdx*.json
 *.code-workspace
 !project.code-workspace
+*~
```

Comment:
Could we please remove any OS-specific ignores, as they are intentionally covered by the [dotfiles](https://github.com/nhs-england-tools/dotfiles/blob/868008b5a7d46b9c44a9ce5a3e5660456b137d5a/dot_gitignore.tmpl#L32) project?

Reply:
I'll go with "yes" because that specific rule is covered (although I'm not sure what else would be generating `*~` files, I've only seen them from emacs)

Reply:
I wasn't aware that Emacs uses `~` for backup files. Based on that, we could probably keep that ignore rule. However, I suspect that not many of us use Emacs. Therefore, the global ignore rules file is a better place for it.